src/pybind/mgr/dashboard_v2/controllers/erasure_code_profile.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+
+from ..services.ceph_service import CephService
+from .. import logger, mgr
+from ..tools import ApiController, AuthRequired, RESTController
+
+
+@ApiController('erasure_code_profile')
+@AuthRequired()
+class ErasureCodeProfile(RESTController):
+
+    def list(self):
+        ret = []
+        for name, ecp in mgr.get('osd_map').get('erasure_code_profiles', {}).items():
+            ecp['name'] = name
+            ret.append(ecp)
+        return ret
+
+    @RESTController.args_from_json
+    def create(self, name, k, m, ruleset_failure_domain=None):
```

Comment:
what about allowing the caller to specify the `plugin` to the be used?

Reply:
I agree, this was something that was brought up in a conversation I had today. Being able to provide the `plugin` would be nice, but then we would also need a method that returns a list of available plugins, so we don't have to maintain a hard-coded list. Not sure if that exists, though. We had similar issues with providing a selection list for compression plugins in openATTIC where we maintained a static list, but the actual build of Ceph may have been compiled with different defaults.

Reply:
fixed